src/user-post/controller/post.controller.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  FileTypeValidator,
+  Get,
+  MaxFileSizeValidator,
+  Param,
+  ParseFilePipe,
+  Patch,
+  Post,
+  Req,
+  UploadedFile,
+  UseInterceptors,
+} from '@nestjs/common';
+import { EditPostDto, PostDto } from '../dto/post.dto/post.dto';
+import { PostService } from '../service/post.service';
+import { Posts } from '../entities/Post.entity';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Request } from 'express';
+import { ApiTags } from '@nestjs/swagger';
+
+@Controller('post')
+@ApiTags('Post')
+export class PostController {
+  constructor(private postService: PostService) {}
+  @Post('')
+  @UseInterceptors(FileInterceptor('photo'))
+  create(
+    @Body() postFormData: PostDto,
+    @UploadedFile(
+      new ParseFilePipe({
+        fileIsRequired: false,
+        validators: [
+          new MaxFileSizeValidator({ maxSize: 10000000 }),
+          new FileTypeValidator({ fileType: '.(png|jpeg|jpg)' }),
+        ],
+      }),
+    )
+    photo: Express.Multer.File,
+  ): Promise<Posts[]> {
+    return this.postService.createPost(postFormData, photo);
+  }
+  @Get('user/:userId')
+  getPosts(@Param('userId') userId: string): Promise<Posts[]> {
+    return this.postService.getPosts(userId);
+  }
+  @Get('/getAllPosts')
+  getAllPosts(): Promise<Posts[]> {
+    console.log('...get all posts');
+    return this.postService.getAllPosts();
+  }
+  @Get('/getAllPosts/:length')
+  getMorePosts(@Param('length') length: number): Promise<Posts[]> {
+    return this.postService.getMorePosts(length);
+  }
+  @Get('/getUserAllPosts/:length')
+  getMoreUserPosts(
+    @Param('length') length: number,
+    @Req() req: Request,
+  ): Promise<Posts[]> {
+    return this.postService.getMoreUserPosts(length, req['user'].user_id);
+  }
+  @Get('/getAllPostsLength')
+  getAllPostsLength(): Promise<number> {
+    return this.postService.getAllPostsLength();
+  }
+  @Get('/getAllUserPostsLength')
+  getAllUserPostsLength(@Req() req: Request): Promise<number> {
+    return this.postService.getAllUserPostsLength(req['user'].user_id);
+  }
+  @Patch('edit/:postId')
+  @UseInterceptors(FileInterceptor('photo'))
+  editPost(
+    @Param('postId') postId: number,
+    @Req() req: Request,
+    @Body() editFormData: EditPostDto,
+    @UploadedFile(
+      new ParseFilePipe({
+        fileIsRequired: false,
+        validators: [
+          new MaxFileSizeValidator({ maxSize: 10000000 }),
+          new FileTypeValidator({ fileType: '.(png|jpeg|jpg)' }),
+        ],
+      }),
+    )
+    photo: Express.Multer.File,
+  ): Promise<[Posts[], string]> {
+    return this.postService.editPost(
+      editFormData,
+      photo,
+      postId,
+      req['user'].user_id,
+    );
+  }
+  @Delete('delete/:postId')
+  deletePost(
+    @Param('postId') postId: number,
+    @Req() req: Request,
+  ): Promise<Posts[]> {
```

Comment:
If we're deleting, then how'll return array of posts?

Reply:
Not really understand what is the question about, after deleting some post i'm return the array that's not containing the old post for front to handle it and dynamically change posts